### PR TITLE
[WIP] Add loading modal during bulk operations

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Towards: #2975
# What's Changed

Screen Shot 2022-12-05 at 4 15 02 PM

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by HanlinMiao** [comment]
This is a part of https://github.com/nautobot/nautobot/issues/2459

File: nautobot/utilities/templates/utilities/templatetags/loading_modal.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        from { -webkit-transform: rotate(0deg); }
+        to { -webkit-transform: rotate(360deg); }
+    }
+</style>
+{% endblock %}
+<div class="modal fade" style="vertical-align: middle;" tabindex="-1" id="loading_modal" data-backdrop="static" aria-labelledby="loadingModalLabel" aria-hidden="true">
+    <div class="modal-dialog modal-lg">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h4 class="modal-title" id="loadingModalLabel">{{action}}ing {{ size }} {{ model_plural_name }} for you, hold on ...</h4>
```
Comment (bryanculver) [inline nautobot/utilities/templates/utilities/templatetags/loading_modal.html:24]:
```suggestion
                <h4 class="modal-title" id="loadingModalLabel">{{action}}ing {{ size }} {{ model_plural_name }}...</h4>
```

**Review by bryanculver — COMMENTED** [review]
Let's see if we can add it to this bulk create path too for full coverage (see: https://github.com/nautobot/nautobot/pull/3008)

pretty neat

File: nautobot/utilities/templates/utilities/templatetags/loading_modal.html (hunk 1 above)
Comment (bryanculver) [reply nautobot/utilities/templates/utilities/templatetags/loading_modal.html:24]: I like it but our current UI/UX doesn't have a lot of personality.

**Comment by gsnider2195** [comment]
If you go to `/dcim/device-types/import/` and enter no data you get a constantly spinning modal:

image

**Comment by gsnider2195** [comment]
Can this also be added to `generic/object_bulk_import.html`? DeviceType is the only model that uses `object_import.html` for its bulk import page.

**Comment by HanlinMiao** [comment]
> 

Yeah I think DeviceType import does not need a loading modal either cause it is only single object. The problem of a constantly spinning modal still remains for CSV file import hmm

**Comment by HanlinMiao** [comment]
> Can this also be added to `generic/object_bulk_create.html`? DeviceType is the only model that uses `object_import.html` for its bulk import page.

Yeah I think DeviceType import does not need a loading modal either cause it is only single object. The problem of a constantly spinning modal still remains for CSV file import hmm

**Pull request closed without merging**